### PR TITLE
Add "Production usage" section to Jedis quick start

### PR DESCRIPTION
**Pull request by uglide**
cc @dwdougherty @sazzad16 @chayim

### CODE REVIEW

File: docs/connect/clients/java.md [hunk 1]
```diff
@@ -187,104 +187,126 @@ public class Main {
 }
 ```
 
-### Example: Indexing and querying JSON documents
+### Production usage
 
-Make sure that you have Redis Stack and `Jedis` installed. 
+### Configuring Connection pool
+As was covered in the previous section, it's recommended to use `JedisPool` or `JedisPooled`. 
+`JedisPooled` is available since Jedis 4.0.0 and is a wrapper around `JedisPool` that provides a simpler API.
```
Comment (sazzad16) [inline docs/connect/clients/java.md:194]: `wrapper` is wrong term here.
It allows similar (exactly same) connection (resource) maintenance as JedisPool.
But, yes, it provides simpler API.

File: docs/connect/clients/java.md [hunk 2]
```diff
@@ -187,104 +187,126 @@ public class Main {
...
+// JedisPooled does all hard work on fetching and releasing connection to the pool
+// to prevent connection starvation
+JedisPooled jedis = new JedisPooled(poolConfig, "localhost", 6379);
 ```
 
-Create an index. In this example, all JSON documents with the key prefix `user:` are indexed. For more information, see [Query syntax](/docs/interact/search-and-query/query/).
 
+### Timeout
+
+To set a timeout for a connection, use `JedisPooled` or `JedisPool` constructor with `timeout` parameter or `JedisClientConfig` with `socketTimeout` parameter:
```
Comment (sazzad16) [inline docs/connect/clients/java.md:247]: There are several types of timeout.
- connectionTimeout
- soTimeout / socketTimeout
- infiniteSoTimeout / blockingSocketTimeout (socketTimeout but for blocking commands)

Lone timeout parameter in constructors acts value for both connectionTimeout and socketTimeout.

File: docs/connect/clients/java.md (hunk 1 above)
Comment (uglide) [reply docs/connect/clients/java.md]: Fixed. Thanks!

File: docs/connect/clients/java.md (hunk 2 above)
Comment (uglide) [reply docs/connect/clients/java.md]: Fixed

**Review by dwdougherty — APPROVED** [review]
LGTM

**Review by sazzad16 — APPROVED** [review]
It suggests Jedis `4.3.1`. Should we do an updated one?
Otherwise, LGTM.

**Comment by sazzad16** [comment]
@uglide BTW, spellcheck fails.

**Pull request merged**